vnext/ReactWindowsCore/Modules/AsyncStorageModuleWin32.cpp

```diff
@@ -166,16 +221,29 @@ AsyncStorageModuleWin32::AsyncStorageModuleWin32() {
     return SQLITE_OK;
   };
 
-  ExecImplThrows(m_db, u8"PRAGMA user_version", getUserVersionCallback, &userVersion);
+  Exec(m_db, u8"PRAGMA user_version", getUserVersionCallback, &userVersion);
 
   if (userVersion == 0) {
-    ExecImplThrows(
+    Exec(
         m_db,
         u8"CREATE TABLE IF NOT EXISTS AsyncLocalStorage(key TEXT PRIMARY KEY, value TEXT NOT NULL); PRAGMA user_version=1");
   }
 }
 
 AsyncStorageModuleWin32::~AsyncStorageModuleWin32() {
+  decltype(m_action) action;
+  {
+    // If there is an in-progress async task, cancel it and wait on the
+    // condition_variable for the async task to acknowledge cancellation by
+    // nulling out m_action. Once m_action is null, it is safe to proceed
+    // wth closing the DB connection
+    winrt::slim_shared_lock_guard guard{m_lock};
+    m_tasks.clear();
```

Comment:
>m_tasks.clear(); [](start = 4, length = 16)

Ideally we should not deallocate object under the lock. The approach below based on std::swap usually considered to be safer.